packages/cli/generators/interceptor/index.js

```diff
@@ -0,0 +1,132 @@
+// Copyright IBM Corp. 2019. All Rights Reserved.
+// Node module: @loopback/cli
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+'use strict';
+const _ = require('lodash');
+const ArtifactGenerator = require('../../lib/artifact-generator');
+const debug = require('../../lib/debug')('interceptor-generator');
+const inspect = require('util').inspect;
+const path = require('path');
+const utils = require('../../lib/utils');
+
+const SCRIPT_TEMPLATE = 'interceptor-template.ts.ejs';
+
+module.exports = class InterceptorGenerator extends ArtifactGenerator {
+  // Note: arguments and options should be defined in the constructor.
+  constructor(args, opts) {
+    super(args, opts);
+  }
+
+  _setupGenerator() {
+    this.option('group', {
+      description: 'Name of the global interceptor group for ordering',
+      required: false,
+      type: String,
+    });
+
+    this.artifactInfo = {
+      type: 'interceptor',
+      rootDir: utils.sourceRootDir,
+    };
+
+    this.artifactInfo.outDir = path.resolve(
+      this.artifactInfo.rootDir,
+      utils.interceptorsDir,
+    );
+
+    this.artifactInfo.defaultTemplate = SCRIPT_TEMPLATE;
+
+    return super._setupGenerator();
+  }
+
+  setOptions() {
+    return super.setOptions();
+  }
+
+  checkLoopBackProject() {
+    if (this.shouldExit()) return;
+    return super.checkLoopBackProject();
+  }
+
+  /**
+   * Ask for Service Name
+   */
+  async promptArtifactName() {
+    debug('Prompting for global interceptor name');
+    if (this.shouldExit()) return;
+
+    if (this.options.name) {
+      Object.assign(this.artifactInfo, {name: this.options.name});
+    }
+
+    const prompts = [
+      {
+        type: 'input',
+        name: 'name',
+        // capitalization
+        message: utils.toClassName(this.artifactInfo.type) + ' name:',
+        when: !this.artifactInfo.name,
+        validate: utils.validateClassName,
+      },
+    ];
+    return this.prompt(prompts).then(props => {
+      Object.assign(this.artifactInfo, props);
+      return props;
+    });
+  }
+
+  async promptInterceptorGroup() {
+    debug('Prompting for global interceptor group');
+    if (this.shouldExit()) return;
+
+    if (this.options.group) {
+      Object.assign(this.artifactInfo, {group: this.options.group});
+    }
+
+    const prompts = [
+      {
+        type: 'input',
+        name: 'group',
+        // capitalization
+        message: utils.toClassName(this.artifactInfo.type) + ' group:',
+        default: '',
```

Comment:
How can the person running `lb4 interceptor` find out what are the valid group names?

What happens when they enter a group name that's not recognized by the app?

Reply:
Based on our design, unknown group is sorted by alphabetical order before the known groups.

Reply:
> Based on our design, unknown group is sorted by alphabetical order before the known groups.

That answered my second question 👍 

The first one is still not answered though:
> How can the person running `lb4 interceptor` find out what are the valid group names?

Reply:
Since the valid group names are configured by a binding, I'm not sure how we can check. Maybe we can improve the prompt to explain so.

Reply:
> Maybe we can improve the prompt to explain so.

Sounds good.

Ideally, I'd like the prompt to tell me:
1. What are the usual (default) group names
2. Where can I find the actual list of group names as recognized by my application